homework/index.js

```diff
@@ -33,14 +36,66 @@
   function main(url) {
     fetchJSON(url, (err, data) => {
       const root = document.getElementById('root');
+
       if (err) {
         createAndAppend('div', root, { text: err.message, class: 'alert-error' });
       } else {
-        createAndAppend('pre', root, { text: JSON.stringify(data, null, 2) });
+        //createAndAppend('pre', root, { text: JSON.stringify(data, null, 2) });
+        const header = createAndAppend('header', root, { class: 'header' });
+        const h1 = createAndAppend('h1', header, { text: 'FooCoding Repos', class: 'h1' });
+
+        const select = createAndAppend('select', root, { class: 'select' });
+        createAndAppend('option', select, { text: 'Choose your favorite repo below:' });
+
+        data.forEach(repo => {
+          const name = repo.name;
+          createAndAppend('option', select, { text: name });
+          //console.log(name);
+        });
+
+        const repoInfo = createAndAppend('div', root, { class: 'left-div' });
+        const contribs = createAndAppend('div', root, { class: 'right-div' });
+        select.addEventListener('change', evt => {
+          const selectedRepo = evt.target.value;
+          const repo = data.filter(r => r.name == selectedRepo)[0]; // ask about this part here
+          console.log(repo.name);
+          repoInfo.innerHTML = repo.name;
+          contribs.innerHTML = repo.name;
+
+          const addInfo = (label, value) => {
+            const container = createAndAppend('div', repoInfo);
+            createAndAppend('span', container, { text: label });
+            createAndAppend('span', container, { text: value });
+          };
+          addInfo('Name: ', repo.name);
+          addInfo('Description: ', repo.description);
+          addInfo('Number of forks: ', repo.forks);
+          addInfo('Updated: ', repo.updated_at);
+          //or instead of using the addInfo function we can use these lines:
+          // createAndAppend('div', repoInfo, { text: `Descriptions: ${repo.description}` });
+          // createAndAppend('div', repoInfo, { text: `Number of Forks: ${repo.forks}` });
+          // createAndAppend('div', repoInfo, { text: `Updated at:${repo.updated_at}` });
+          const contribsUrl = repo.contributors_url;
+          fetchJSON(contribsUrl, (err, contribData) => {
+            if (err) {
+              createAndAppend('div', root, { text: err.message, class: 'alert-error' });
+            } else {
+            }
+            contribData.forEach(contributor => {
+              createAndAppend('div', contribs, { text: contributor.login });
+              //createAndAppend('a', contribs, { href: contributor.html.url, setAttribute: 'href' });
```

Comment:
I think the issues here are:

 1. `contributor.html.url` should be `contributor.html_url`
 2. you need to add `text: contributor.login`

Reply:
So that's why it wasn't working. Thanks!